src/languageFacts/entry.ts

```diff
@@ -141,32 +196,34 @@ function getEntryMarkdownDescription(entry: IEntry2, settings?: HoverSettings):
 	return result;
 }
 
+// TODO: Remove "as any" when tsconfig supports es2021+
+const missingBaselineBrowserFormatter = new (Intl as any).ListFormat("en", {
+	style: "long",
+	type: "disjunction",
+});
+
 /**
- * Input is like `["E12","FF49","C47","IE","O"]`
- * Output is like `Edge 12, Firefox 49, Chrome 47, IE, Opera`
+ * Input is like {'chrome': '93', 'edge': '93', 'firefox': '92'}
+ * Output is like `Safari`
  */
-export function getBrowserLabel(browsers: string[] = []): string | null {
-	if (browsers.length === 0) {
-		return null;
-	}
-
-	return browsers
-		.map(b => {
-			let result = '';
-			const matches = b.match(/([A-Z]+)(\d+)?/)!;
-
-			const name = matches[1];
-			const version = matches[2];
-
-			if (name in browserNames) {
-				result += browserNames[name as keyof typeof browserNames];
-			}
-			if (version) {
-				result += ' ' + version;
-			}
-			return result;
-		})
-		.join(', ');
+export function getMissingBaselineBrowsers(support?: BaselineSupport): string {
+	if (!support) {
+		return '';
+	}
+	const missingBrowsers = new Map(Object.entries(browserNames));
+	for (const browser in support) {
+		missingBrowsers.delete(browser);
+	}
+
+	return missingBaselineBrowserFormatter.format(Object.values(Array.from(missingBrowsers.entries()).reduce((browsers: Record<string, string>, [browserId, browser]) => {
+		if (browser.name in browsers || browserId == 'edge') {
```

Comment:
The core browser set is not likely to change much, but if it does, this hard cording (as well as the `browserNames` const are a bit unfortunate as they'll need to be changed if the core browser set changes.
Can you think of a utility package/function which we could maintain on the web-features repo instead, which would avoid consumers like this one to have to do this?

Reply:
Hm actually the `computeBaseline` function does already expose browser names and platforms, eg:

```
Browser {
    id: 'chrome',
    data: {
      accepts_flags: true,
      accepts_webextensions: true,
      name: 'Chrome',
      pref_url: 'chrome://flags',
      preview_name: 'Canary',
      releases: [Object],
      type: 'desktop'
    },
    ...
}
```

I'll look into moving the "missing browser" functionality over to [vscode-custom-data#111](https://github.com/microsoft/vscode-custom-data/pull/111/files#diff-df39d64e75bdd56b21124ae2fcdc9778ae695292a82c50630bd1ebfdbc9e9f2aR374) where it already uses `computeBaseline`.

Reply:
Given @aeschli's [comment](https://github.com/microsoft/vscode-custom-data/pull/111#issuecomment-2740307413) about the need for a well-defined schema, I think this is actually ok. If the core browser set changes, this implementation will need to be updated in a few places anyway.